Microsoft.Toolkit.Uwp.UI.Controls.DataGrid/DataGrid/DataGridComboBoxColumn.cs

```diff
@@ -0,0 +1,547 @@
+        }
+
+        /// <summary>
+        /// Gets or sets the font size.
+        /// </summary>
+        // Use DefaultValue here so undo in the Designer will set this to NaN
+        [DefaultValue(double.NaN)]
+        public double FontSize
+        {
+            get
+            {
+                return _fontSize ?? double.NaN;
+            }
+
+            set
+            {
+                if (_fontSize != value)
+                {
+                    _fontSize = value;
+                    NotifyPropertyChanged(DATAGRIDCOMBOBOXCOLUMN_fontSizeName);
+                }
+            }
+        }
+
+        /// <summary>
+        /// Gets or sets the font style.
+        /// </summary>
+        public FontStyle FontStyle
+        {
+            get
+            {
+                return _fontStyle ?? FontStyle.Normal;
+            }
+
+            set
+            {
+                if (_fontStyle != value)
+                {
+                    _fontStyle = value;
+                    NotifyPropertyChanged(DATAGRIDCOMBOBOXCOLUMN_fontStyleName);
+                }
+            }
+        }
+
+        /// <summary>
+        /// Gets or sets the font weight or thickness.
+        /// </summary>
+        public FontWeight FontWeight
+        {
+            get
+            {
+                return _fontWeight ?? FontWeights.Normal;
+            }
+
+            set
+            {
+                if (!_fontWeight.HasValue || _fontWeight.Value.Weight != value.Weight)
+                {
+                    _fontWeight = value;
+                    NotifyPropertyChanged(DATAGRIDCOMBOBOXCOLUMN_fontWeightName);
+                }
+            }
+        }
+
+        /// <summary>
+        /// Gets or sets a brush that describes the foreground of the column cells.
+        /// </summary>
+        public Brush Foreground
+        {
+            get
+            {
+                return _foreground;
+            }
+
+            set
+            {
+                if (_foreground != value)
+                {
+                    _foreground = value;
+                    NotifyPropertyChanged(DATAGRIDCOMBOBOXCOLUMN_foregroundName);
+                }
+            }
+        }
+
+        /// <summary>
+        /// Gets a <see cref="T:Windows.UI.Xaml.Controls.ComboBox"/> control that is bound to the column's ItemsSource collection.
+        /// </summary>
+        /// <param name="cell">The cell that will contain the generated element.</param>
+        /// <param name="dataItem">The data item represented by the row that contains the intended cell.</param>
+        /// <returns>A new <see cref="T:Windows.UI.Xaml.Controls.ComboBox"/> control that is bound to the column's ItemsSource collection.</returns>
+        protected override FrameworkElement GenerateEditingElement(DataGridCell cell, object dataItem)
+        {
+            var value = dataItem.GetType().GetProperty(Binding.Path.Path).GetValue(dataItem);
```

Comment:
In this case I would like to throw an exception and inform the user that they forgot to specify the binding. What do you think?

Reply:
Here's my point of view Filip: 
I can bind a DataGrid to a data source, and then do
    myDataGrid.Columns.Add(new DataGridTextColumn());
without exception. The column gets added with empty cells.
Later I can dynamically set the DataGridBoundColumn.Binding property and the cells get populated.

But I could not do
    myDataGrid.Columns.Add(new DataGridComboBoxColumn());
because of that exception.

My preference would be to be able to do the same with the ComboBox column type.

Notice how OnColumnBindingChanged gets called when a new DataGridBoundColumn.Binding is set.    Thanks!

Reply:
I can make that happen later if you prefer.